src/main/java/com/google/devtools/build/lib/bazel/bzlmod/BazelLockFileModule.java

```diff
@@ -123,7 +124,7 @@ private ImmutableMap<ModuleExtensionId, LockFileModuleExtension> combineModuleEx
     for (ModuleExtensionResolutionEvent extensionEvent : extensionResolutionEvents) {
       updatedExtensionMap.put(extensionEvent.getExtensionId(), extensionEvent.getModuleExtension());
     }
-    return updatedExtensionMap.buildKeepingLast();
+    return ImmutableSortedMap.copyOf(updatedExtensionMap.buildKeepingLast());
```

Comment:
I believe this entire PR can be reduced to this one line. (well, and using the lexicographical comparator explicitly.) ImmutableMap already has a stable iteration order so we don't need to hold on to it as a sorted map (especially as the iteration order is only ever important for serialization).

Reply:
Well, that's much nicer now, thanks!